waltid-services/waltid-issuer-api/src/main/kotlin/id/walt/issuer/issuance/CIProvider.kt

```diff
@@ -234,56 +246,73 @@ open class CIProvider(
         log.debug { "RETRIEVING ISSUANCE REQUEST FOR CREDENTIAL REQUEST" }
         val request = findMatchingIssuanceRequest(
             credentialRequest, issuanceSession.issuanceRequests
-        ) ?: throw IllegalArgumentException("No matching issuance request found for this session: ${issuanceSession.id}!")
+        )
+            ?: throw IllegalArgumentException("No matching issuance request found for this session: ${issuanceSession.id}!")
 
         return CredentialResult(format = credentialRequest.format, credential = JsonPrimitive(runBlocking {
             val vc = request.credentialData ?: throw MissingFieldException(listOf("credentialData"), "credentialData")
             val resolvedIssuerKey = KeyManager.resolveSerializedKey(request.issuerKey)
 
             request.run {
-                val holderKeyJWK =  JWKKey.importJWK(holderKey.toString()).getOrNull()?.exportJWKObject()?.plus("kid" to JWKKey.importJWK(holderKey.toString()).getOrThrow().getKeyId())?.toJsonObject()
+                val holderKeyJWK = JWKKey.importJWK(holderKey.toString()).getOrNull()?.exportJWKObject()
+                    ?.plus("kid" to JWKKey.importJWK(holderKey.toString()).getOrThrow().getKeyId())?.toJsonObject()
 
                 when (credentialFormat) {
                     CredentialFormat.sd_jwt_vc -> OpenID4VCI.generateSdJwtVC(
+
                         credentialRequest = credentialRequest,
                         credentialData = vc,
                         issuerId = issuerDid ?: baseUrl,
                         issuerKey = resolvedIssuerKey,
                         selectiveDisclosure = request.selectiveDisclosure,
                         dataMapping = request.mapping,
-                        display = credentialRequest.display,
+                        display = if (credentialRequest.display != null) credentialRequest.display else vc.jsonObject["display"]?.jsonArray?.map {
```

Comment:
this could also be written in a more 'kotlin' way, e.g.
```kotlin
display = credentialRequest.display ?: vc["display"]?.jsonArray?.map {
    DisplayProperties.fromJSON(it.jsonObject)
},
```